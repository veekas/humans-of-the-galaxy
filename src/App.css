.App {
  text-align: center;
}

.App-header {
  background-color: #131313;
  padding: 20px;
  color: white;
}

.App-title {
  font-family: 'Gill Sans', 'Gill Sans MT', Calibri, 'Trebuchet MS', sans-serif;
  font-size: 1.5em;
}

.App-intro {
  font-size: large;
}

ul {
  display: inline;
}
