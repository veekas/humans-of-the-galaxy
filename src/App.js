import React, { Component } from 'react';

import { AppContainer, Header, Title, MainLoading, MainContainer } from './styled/appElements';
import { UnorderedList } from './styled/listElements';

import Planet from './Planet';
import PageControls from './PageControls';

import { getPlanets } from './api';

class App extends Component {
  state = {
    planets: [],
    loading: true,
  }

  componentDidMount = () => {
    getPlanets().then(planets => {
      this.setState({
        planets,
        loading: !this.state.loading,
      });
    });
  }

  render() {
    const { planets, loading } = this.state;

    return (
      <AppContainer>
        <Header>
          <Title>People of Star Wars</Title>
        </Header>

        <MainContainer>
          <UnorderedList>
            {
              !loading && planets.length
                ?
                planets.map(planet => (
                  <Planet
                    key={planet.name}
                    {...planet}
                  />
                ))
                : <MainLoading src="loading.svg" alt="Loading..." />
            }
          </UnorderedList>
        </MainContainer>

        <PageControls />
      </AppContainer>
    );
  }
}

export default App;
