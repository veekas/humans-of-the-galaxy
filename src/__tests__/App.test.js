import React from 'react';
import ReactDOM from 'react-dom';
import { create } from 'react-test-renderer';
import App from '../App';

describe('<App>', () => {
  it('renders without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(<App />, div);
    ReactDOM.unmountComponentAtNode(div);
  });

  it('matches snapshot on shallow render', () => {
    const component = create(<App />);
    const snap = component.toJSON();
    expect(snap).toMatchSnapshot();
  });
});
