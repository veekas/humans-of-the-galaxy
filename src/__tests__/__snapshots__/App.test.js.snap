// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<App> matches snapshot on shallow render 1`] = `
<div
  className="sc-htpNat eZrIBQ"
>
  <header
    className="sc-bxivhb jDSttw"
  >
    <h1
      className="sc-ifAKCX iLUShu"
    >
      People of The Galaxy
    </h1>
  </header>
  <div
    className="sc-bZQynM gRuQiS"
  >
    <img
      alt="Loading..."
      className="sc-EHOje hbHrld"
      src="loading.svg"
    />
  </div>
  <footer
    className="sc-gzVnrw kgUxTz"
  >
    <img
      className="sc-htoDjs dSpAyG"
      onClick={[Function]}
      src="back.svg"
    />
    <h1
      className="sc-ifAKCX iLUShu"
    >
      Page 
      1
       of 7
    </h1>
    <img
      className="sc-htoDjs dSpAyG"
      onClick={[Function]}
      src="next.svg"
    />
  </footer>
</div>
`;
